src/components/search/SearchSection.jsx

```diff
@@ -1,35 +1,81 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect, useContext } from 'react';
+import axios from 'axios';
 import { FaSearch, FaMapMarkerAlt, FaRegCalendar } from 'react-icons/fa';
 import DatePicker from 'react-datepicker';
-import AutoComplete from './Autocomplete';
+// import AutoComplete from './Autocomplete';
 import 'react-datepicker/dist/react-datepicker.css';
 import styles from '../../styles/Search.module.css';
+import SearchContext from './SearchContext';
 
 const SearchSection = () => {
   const dummyTags = ['Starting soon', 'Tomorrow', 'Near you', 'Dancers', 'Clowns', 'Magicians'];
+  const geoLocation = { lat: 29.954767355989652, lng: -90.06911208674771 };
   const [searchTerm, setSearchTerm] = useState('');
-  const [searchLocation, setSearchLocation] = useState('');
+  const [searchLocation, setSearchLocation] = useState(geoLocation);
   const [searchDate, setSearchDate] = useState('');
+  const { setResults } = useContext(SearchContext);
+
+  const onSearchSubmit = () => {
+    axios.get('https://www.buskr.life/api/events', {
+      params: {
+        features: 'coords,location,photos,tags',
+        lat: searchLocation.lat,
+        lng: searchLocation.lng,
+        from: searchDate
+          || new Date(),
+        limit: 100,
+      },
+    }).then((result) => {
+      const byTime = result.data.features.sort(
+        (a, b) => new Date(a.properties.starts) - new Date(b.properties.starts),
+      );
+
+      let bySearchTerm = byTime;
+      if (searchTerm) {
+        bySearchTerm = result.data.features.filter(
+          (event) => {
+            const filteredEvents = event.properties.name.toLowerCase()
+              .includes(searchTerm.toLowerCase())
+              || event.properties.buskerName.toLowerCase().includes(searchTerm.toLowerCase())
+              || event.properties.tags.indexOf(searchTerm.toLowerCase()) !== -1;
+            return filteredEvents;
+          },
+        );
+      }
+      setResults({ results: byTime, filtered: bySearchTerm });
```

Comment:
I think by default, people would prefer to receive the results sorted by distance, which is how they arrive from the server, rather than by time. Or you could put both in there. But it should definitely be possible to access a list of results sorted by distance, not time.

Reply:
I made it so that you can access a list of results sorted by distance, time, and keyword for a single date